C#/C#.sublime-syntax

```diff
@@ -101,13 +103,17 @@ contexts:
       push:
         - meta_scope: comment.block.source.cs
         - match: \*/\n?
+          captures:
+            0: punctuation.definition.comment.source.cs
           pop: true
     - match: //
       captures:
         1: punctuation.definition.comment.source.cs
       push:
         - meta_scope: comment.line.double-slash.source.cs
         - match: $\n?
+          captures:
+            1: punctuation.definition.comment.source.cs
```

Comment:
This looks like nonsense to me - there is no capture `1` in the above string

Reply:
if you look at the PR text, you can see it was just a conversion from `tmLanguage`, and it was so long ago that presumably its not even relevant now? Did you check on `master`?

Reply:
On master this type of mistake is still present in some places. Perhaps it should be an error to specify a `captures` index which doesn't exist?

Reply:
You are correct in that this PR added some superfluous captures, but they are an artifact of the conversion and technically required for feature parity. It's just that in some cases (like this one here) the end pattern (now a pop-match) didn't actually use the `captures` of the pattern. If you'd like to go ahead and remove those that still exist, that'd be appreciated, but as it is they don't do anything bad.

Reply:
That is unsurprising as that syntax is practically untouched since the conversion, no? Sometimes capture groups exist in the pattern but don't participate in a match, so if this were to be implemented it would have to work at regex compile time and not during lexing (probably I'm stating the obvious here, but just in case ;))

Reply:
> Sometimes capture groups exist in the pattern but don't participate in a match

That's not the case I'm suggesting become an error. Referring to group 1 in `(?:matches|(doesn't) match)` is fine, but referring to group 2 in that regex is knowably stupid at regex compilation/lex-time before attempting to match the regex against any text.

Reply:
>That's not the case I'm suggesting become an error / warning.

What are you suggesting then?

I was just explaining that captures referring to groups that don't exist (in pop matches) is an artifact of the old tmLanguage format (explained [on the unofficial docs](http://docs.sublimetext.info/en/latest/extensibility/syntaxdefs.html)) where `captures` applied to both begin and end patterns if no `beginCaptures` or `endCaptures` mapping has been provided. (I don't think this is documented.)

I'd be perfectly fine with ST reporting these as warnings in the console on compilation, as they hint at a user error more than anything. Regardless, that should be a suggestion made on https://github.com/SublimeTextIssues/Core.